server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -165,14 +165,26 @@ abstract class AbstractSearchAsyncAction<Result extends SearchPhaseResult> exten
         this.clusters = clusters;
     }
 
+    // we need to add 1 for non-active partition, since we count it in the total. This means for each shard in the iterator we sum up
+    // it's number of active shards but use 1 as the default if no replica of a shard is active at this point.
+    // on a per shards level we use shardIt.remaining() to increment the totalOps pointer but add 1 for the current shard result
+    // we process hence we add one for the non-active partition here.
+    private static int totalSizeWith1ForEmpty(List<? extends Countable> iterators) {
```

Comment:
I thought we got rid of this, we should be able to remove it, right?

Reply:
Right missed this one :)